Docs: Instructions on how to write E2E tests for app plugins (#891)
grafana/plugin-tools: docusaurus/docs/e2e-test-a-plugin/test-an-app-plugin/app-pages.md

import pluginJson from '../src/plugin.json';

type AppTestFixture = {
  gotoPage: (path?: string) => Promise<AppPage>;
};

export const test = base.extend<AppTestFixture>({
  gotoPage: async ({ gotoAppPage }, use) => {
    await use((path) =>
      gotoAppPage({
        path,
        pluginId: pluginJson.id,
      })
    );
  },
});

export { expect } from '@grafana/plugin-e2e';
```

To use this function, simply import `test` and `expect` from your fixture instead of importing it from `@grafana/plugin-e2e`, and then write your test as you would normally.

For example:

```ts title="startPage.spec.ts"
import { test, expect } from './fixtures.ts';

test('start page should welcome users to the app', async ({ gotoPage, page }) => {
  await gotoPage('/start');
  await expect(page.getByRole('heading', { name: 'Welcome to my app' })).toBeVisible();
});
```

## Page with a complex UI

If, on the other hand, you have a page with a complex UI that might not be easy to interact with via the standard Playwright APIs, then you need a different approach. As a best practice, create a page object that can encapsulate the page's logic in functions.

> This is especially helpful if you want to reuse that selector logic in multiple tests. The example below showcases how to extend the standard fixture with a page object. Simplify the `getWelcomeText` logic because you want to highlight the page object pattern rather than write a complex selector logic.

Author: ```suggestion
This is especially helpful if you want to reuse that selector logic in multiple tests. The example below showcases how to extend the standard fixture with a page object. We have simplified the `getWelcomeText` logic to highlight the page object pattern without introducing unnecessary complexity to this example.
```

Author: What do you think about something like this @josmperez
